server/index.js

```diff
@@ -133,10 +133,11 @@ app.use('/api/comments/byepisodeid', async (req, res) => {
 
     const socialInteract = response.episode.socialInteract && response.episode.socialInteract.filter((si) => si.protocol === 'activitypub');
 
-    if(!socialInteract && socialInteract.lenght >= 0) {
+    if(socialInteract === undefined || socialInteract.length >= 0) {
```

Comment:
```suggestion
    if(socialInteract === undefined || socialInteract.length === 0) {
```

The original condition was broken in more then just the typo, I think, but if I not mistaken, it was designed for: 
1. Returning 400 when the episode does not have a socialInteract (at all)
2. Returning 400 when the episode has one or more socialInteract, but none is `activitpub` (possible, even though unlikely at the moment)

With the code you proposed, the valid use case (episodes with activity pub comments) would not work, as socialInteract.length >= 0 would be true, thus also triggering the 400 error.

I tested the suggestion in this comment and it worked with both episodes with and without comments.
I could not test the scenario described with 2, as I am not aware of any podcast with that use case. I would need to fabricate one, and that would be time consuming.